scene/2d/gpu_particles_2d.cpp

```diff
@@ -334,6 +414,11 @@ void GPUParticles2D::restart() {
 }
 
 void GPUParticles2D::_notification(int p_what) {
+#ifdef TOOLS_ENABLED
+	if (p_what == NOTIFICATION_READY) {
+		_material_changed();
+	}
```

Comment:
It should probably check `Engine::get_singleton()->is_editor_hint()`, otherwise this will also run when you play the game through the editor.